tests/integration/dedicated_alb/test_dedicated_alb_provisioning.py

```diff
@@ -0,0 +1,332 @@
+import json
+from datetime import date
+
+import pytest  # noqa F401
+
+from broker.extensions import config, db
+from broker.models import (
+    Challenge,
+    Operation,
+    DedicatedALBServiceInstance,
+    DedicatedALBListener,
+)
+from broker.tasks.alb import get_lowest_used_alb
+
+from tests.lib.factories import ALBServiceInstanceFactory
+from tests.lib.client import check_last_operation_description
+
+from tests.integration.dedicated_alb.test_dedicated_alb_update import (
+    subtest_update_happy_path,
+    subtest_update_noop,
+)
+
+# The subtests below are "interesting".  Before test_provision_happy_path, we
+# had separate tests for each stage in the task pipeline.  But each test would
+# have to duplicate much of the previous test.  This was arduous and slow. Now
+# we have a single test_provision_happy_path, and many subtest_foo helper
+# methods.  This still makes it clear which stage in the task pipeline is
+# failing (look for the subtask_foo in the traceback), and allows us to re-use
+# these subtasks when testing failure scenarios.
+
+
+def test_provision_happy_path(
```

Comment:
I think my main question is: why can't we test by invoking the huey pipeline for provisioning?

Reply:
To be more precise, why can't we use Huey itself to run the pipeline in the test?

Reply:
under the hood, we are using huey to run the pipeline, but we're forcing it to only grab one task at a time (we do this [here](https://github.com/cloud-gov/external-domain-broker/blob/cc05cf3341dc7e774e97257c950d4db78ed36a9a/tests/lib/tasks.py#L52-L76))
if you're asking why we don't just hand Huey the whole pipeline and check on it at the end:
- I think it's a lot easier to track our mocks/fixtures/etc if we handle them one step at a time; if we did the whole pipeline, we'd have to queue up every expected AWS call up front 
- having the subtests lets us test idempotency of each task as we go
- it's easier to assure that we're doing the right things and in our correct state all the way down